ras-a - use a proper range for messages

hamishwillee opened the pull request:
RAS-A needs a proper allocated range for messages and for commands. Otherwise you are pretty much making it impossible to interwork with non-compliant systems in the network in future (you might not think this could ever happen, but it could).

Right now there are no clashes, but the selected ids for messages pretty much guarantee that there will be in future. There is really no way you can avoid that other than strictly using an allocated range OR including the xml in upstream mavlink.

EDITED: Have reserved range of 51000 - 52000 for message ids and for command ids.

I am aware this will break everything. Better now than later.

@TSC21 !

hamishwillee commented on message_definitions/v1.0/ras_a.xml:
```diff
@@ -384,7 +384,7 @@
     </enum>
   </enums>
   <messages>
-    <message id="238" name="POI_REPORT">
```
This one has to change - no one is allowed to use 0-256 as this is the original mavlink1 range, and is needed for the very rare cases where we need to have a mavlink 1 message. This is guaranteed to clash in future as there are not many unused values left.

hamishwillee commented on message_definitions/v1.0/ras_a.xml:
```diff
@@ -423,7 +423,7 @@
       <field type="float" name="vel_d" units="m/s" invalid="NaN">Down velocity of the POI. NAN if unknown.</field>
       <field type="float" name="hdg" units="rad" invalid="NaN">Heading of the POI in the NED frame. NAN if unknown.</field>
     </message>
-    <message id="450" name="EXPLORATION_STATUS">
```
These are in the range for common.xml (and development.xml while being reviewed). We will be using this range in future.

TSC21 reviewed (approved):
@hamishwillee thanks so much for bringing this up! This makes total sense and was not considered while extending the message set on the new dialect. For me this is good, but have to bring it with the larger group. Thanks again!

TSC21 commented:
For commands: have not evaluated any range yet. Want yo suggest one?

hamishwillee commented:
> For commands: have not evaluated any range yet. Want yo suggest one?

I'm just writing a parser to generate a list of the used ranges so I don't give you poor direction. Once I've confirmed all the ranges I'll check against what you've used to see if we can leave any alone.

I'm going to suggest we add range values to the XML and provide tools to warn users. That will encourage ranges in the first place and make it easier for people coming along later to do the right thing. Won't happen for a bit though.

hamishwillee commented:
OK, so the command ranges are: 
-  ASLUAV.xml range of IDs: commands: 40001 - 41999
- ArduPilot starts at  42001
- 
RasA commands are 40500 to 41051. So they all have to move. 

I can see the highest allocated range is     commands: 50000 - 50099 (herelink). So how about we allocate RasA initially 51000 to 52000?

hamishwillee commented on message_definitions/v1.0/ras_a.xml:
```diff
@@ -384,7 +384,7 @@
   </enums>
   <messages>
-    <message id="238" name="POI_REPORT">
+    <message id="30000" name="POI_REPORT">
```
```suggestion
    <message id="51000" name="POI_REPORT">
```

hamishwillee commented on message_definitions/v1.0/ras_a.xml:
```diff
@@ -423,7 +423,7 @@
       <field type="float" name="hdg" units="rad" invalid="NaN">Heading of the POI in the NED frame. NAN if unknown.</field>
     </message>
-    <message id="450" name="EXPLORATION_STATUS">
+    <message id="30100" name="EXPLORATION_STATUS">
```
```suggestion
    <message id="51100" name="EXPLORATION_STATUS">
```

hamishwillee commented on message_definitions/v1.0/ras_a.xml:
```diff
@@ -435,7 +435,7 @@
       <field type="int8_t" name="level" invalid="INT8_MAX">In an indoor exploration task, it indicates the floor/level of the structure that is currently being explored. The level where the vehicle started the exploration is considered the level 0. INT8_MAX when unknown, not capable to provide or not applicable.</field>
     </message>
-    <message id="451" name="EXPLORATION_INFO">
+    <message id="30101" name="EXPLORATION_INFO">
```
```suggestion
    <message id="51101" name="EXPLORATION_INFO">
```

hamishwillee commented on message_definitions/v1.0/ras_a.xml:
```diff
@@ -466,7 +466,7 @@
       <field type="float" name="egress_portal_alt" units="m" invalid="NaN">Currently defined egress portal Altitude (MSL). NaN if unknown, not applicable or when egress_portal_id is set and different from UINT32_MAX.</field>
     </message>
-    <message id="452" name="EXPLORATION_RETURN_POSITION">
+    <message id="30102" name="EXPLORATION_RETURN_POSITION">
```
```suggestion
    <message id="51102" name="EXPLORATION_RETURN_POSITION">
```

hamishwillee commented on message_definitions/v1.0/ras_a.xml:
```diff
@@ -1,5 +1,9 @@
 <mavlink>
+  <!-- Contact info: ? -->
+  <!-- URL: ? -->
+  <!-- message ID range: 51000 - 52000 -->
+  <!-- command ID range: 51000 - 52000 -->
```
FYI, I have allocated message IDs and command IDs to have the same range to make management a little easier. If this ends up in mavlink/mavlink then you'd need to add contact info and URL - essentially so we know who to contact for changes.

If these changes are made then I'll add a note in mavlink/mavlink to reserve the range.

TSC21 replied:
@hamishwillee please do reserve the range in `mavlink/mavlink`. Thanks!

TSC21 replied:
> add contact info and URL - essentially so we know who to contact for changes.

Feel free to add me:

```xml
<!-- Contact: Nuno Marques <n.marques21@hotmail.com> | Github: TSC21 -->
```

hamishwillee commented:
Any discussion on this yet?

TSC21 commented:
> Any discussion on this yet?

@hamishwillee not yet but I don't see a problem on this. I will let you know as soon as there is something (potentially tomorrow, as we are having a flight control interface meeting). @ThomasDebrunner @tstastny FYI.

hamishwillee commented:
Great, am waiting with baited breath :-)

TSC21 commented:
@radarku @uasclark @thomas-watters-skydio @dskyle-shieldai this directly affects your implementations. Please give your thumbs-up so this can get in. Thanks!

uasclark commented:
This looks necessary and reasonable to me. That is easy to say from my perspective because we directly use generated pymavlink code and do not expect any manual changes as a result.

radarku commented:
This seems prudent theoretically, but practically, we would need @dskyle-shieldai to agree to making the change and also update our platforms.

hamishwillee commented on message_definitions/v1.0/ras_a.xml:
```diff
@@ -1,5 +1,9 @@
 <?xml version="1.0"?>
 <mavlink>
+  <!-- Contact info: ? -->
+  <!-- URL: ? -->
+  <!-- message ID range: 51000 - 52000 -->
+  <!-- command ID range: 51000 - 52000 -->
```
```suggestion
  <!-- Contact: Nuno Marques <n.marques21@hotmail.com> | Github: TSC21 -->
  <!-- message ID range: 51000 - 51999 -->
  <!-- command ID range: 51000 - 51999 -->
```

hamishwillee commented:
@TSC21 Cool. I dropped the range by 1 to 51999 for symmetry and added your contact details. I guess you'll have to merge this to make the change "real".

In mavlink land I added https://github.com/mavlink/mavlink/pull/1957 to record the range in all.xml

TSC21 commented:
Merging as this was approved by the required people.

TSC21 commented:
> @TSC21 Cool. I dropped the range by 1 to 51999 for symmetry and added your contact details. I guess you'll have to merge this to make the change "real".
> 
> In mavlink land I added [mavlink#1957](https://github.com/mavlink/mavlink/pull/1957) to record the range in all.xml

Thanks @hamishwillee!
